### PR TITLE
use https for some gnu.org, nongnu.org urls

### PR DESCRIPTION
**Pull request by chrstphrchvz**
Left alone any links in \*/\*/files/\* (patches, etc.) 

I didn't update url in master_sites of `fontforge` to prevent conflict with #1674. The link is to be removed by that PR, but I can include the change if desired.

The only already-broken links I've spotted so far are those containing `releases-noredirect`.

Known working https gnu.org subdomains:

- download.savannah
- ftp
- gcc 
- lists
- savannah
- www

Known working https nongnu.org subdomains:

- download.savannah (HTTPS Everywhere doesn't have this one yet&hellip;)
- savannah
- www

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

~~macOS 10.x~~
~~Xcode 8.x~~

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] ~~referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?~~

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by raimue** [comment]
This probably breaks legacy systems without support for TLS 1.2, see https://trac.macports.org/ticket/51516. No objection from me, though, just mentioning the fact.

**Comment by Schamschula** [comment]
I'm fine with the change to libtasn1.

**Comment by pmetzger** [comment]
LGTM.

On the older systems that don't do TLS 1.2 problem, I think we will need a solution for them either way and this shouldn't be blocked by that, we should just note it continues to be an issue.

**Comment by pmetzger** [comment]
We will need to wait on committing this until the full three days are up I think.

**Comment by g5pw** [comment]
:+1: for `avrdude`

**Comment by yan12125** [comment]
I think TLS 1.2-only sites no longer bring issues to legacy systems with distfiles mirrors set up?

**Comment by seanfarley** [comment]
Sure, LGTM

File: _resources/port1.0/livecheck/gnu.tcl [hunk 1]
```diff
@@ -1,14 +1,14 @@
 # This file contains the defaults for gnu.
 
 if {${livecheck.name} eq "default"} {
-    if {$has_homepage && [regexp {^http://www.gnu.org/software/([^/]+)} $homepage _ tag]} {
+    if {$has_homepage && [regexp {^https://www.gnu.org/software/([^/]+)} $homepage _ tag]} {
```
[the remark this answers is not in the document]
Comment (jmroot) [reply _resources/port1.0/livecheck/gnu.tcl]: It's setting a default value for livecheck.name if that can be derived from the homepage. This is shared code that is used by ports that aren't even in the official tree, so assume as little as possible.

[the remark this answers is not in the document]
Comment (pmetzger) [reply _resources/port1.0/livecheck/gnu.tcl]: Just for clarity, does this mean that I can or cannot merge this tomorrow?

**Comment by pmetzger** [comment]
I currently intend to merge this tomorrow unless otherwise informed.

File: _resources/port1.0/livecheck/gnu.tcl (hunk 1 above)
[the remark this answers is not in the document]
Comment (jmroot) [reply _resources/port1.0/livecheck/gnu.tcl]: If merging means everyone forgets about this and never fixes it, then please don't.

[the remark this answers is not in the document]
Comment (pmetzger) [reply _resources/port1.0/livecheck/gnu.tcl]: Let me put it differently: I didn't understand what you were suggesting from your earlier comment, as it wasn't explicit enough. Can you be more precise?

[the remark this answers is not in the document]
Comment (jmroot) [reply _resources/port1.0/livecheck/gnu.tcl]: Yes, it should match either http or https.

[the remark this answers is not in the document]
Comment (pmetzger) [reply _resources/port1.0/livecheck/gnu.tcl]: So it should be `^https?:`, correct?

If that is so, @jmroot, then @chrstphrchvz should make that change before we merge.

[the remark this answers is not in the document]
Comment (chrstphrchvz) [reply _resources/port1.0/livecheck/gnu.tcl]: Changed to `^https?://www\.gnu\.org/software/([^/]+)` in 39548997a9. Tested by commenting out the `else`.

**Comment by pmetzger** [comment]
Merged. Thank you, @chrstphrchvz!

**Pull request merged**